cmd/create.go

```diff
@@ -63,7 +64,8 @@ kn func create --template events myfunc
 
 	cmd.Flags().BoolP("confirm", "c", false, "Prompt to confirm all configuration options (Env: $FUNC_CONFIRM)")
 	cmd.Flags().StringP("runtime", "l", fn.DefaultRuntime, "Function runtime language/framework. Available runtimes: "+buildpacks.Runtimes()+" (Env: $FUNC_RUNTIME)")
```

Comment:
This gets a bit tricky doesn't it? The languages available are part of the directory tree when `-r` or `-g` are used. For example, in [this example](https://github.com/lance/gcf-kn) repository the only runtime available is `ruby`. Is a user expected to know this? I'm not sure that we should print available runtimes by default given that extensible and remote language packs are probably going to become much more widely used with the move upstream.

This makes me wonder if we should be thinking a little bigger than a new flag. Just brainstorming here, so...

Should we consider that support for remote repositories also includes a new mechanism for managing the default repository used on all commands? For example:

```
❯ kn func config --global
? What do you want to configure?  [Use arrows to move, type to filter]
> Language Pack Repositories
```
Which would allow you to `List`, `Add` and `Remove`, and `Set Default`. `List` shows `Builtin`, and then lists top level directories in `$HOME/.config/func/repositories`. `Add` accepts a path to a local directory which is copied (symlinked?) into `$HOME/.config/func/repositories`, or it accepts a Git repository URI, and downloads/clones that repository into the repositories directory. `Remove` deletes the repository from the configuration directory, and `Set Default` writes to `$HOME/.config/func/config.yaml`.

I know this would  make the PR much bigger :fearful: 

But it might be a nice experience for both the user and vendors. And it would allow us to display the available runtimes for `-l` dynamically based on what is the user default. Initially those are the ones available builtin.  It also eliminates the need for the `-r` flag - or changes it to accept the name of any top level directory in `$HOME/.config/func/repositories`.

Sorry! This little detail had been bothering me, and I didn't realize it was going to such a long comment...

WDYT?

Reply:
I meant to point out something before I got carried away. If vendors bundle their language packs like this https://github.com/lance/gcf-kn where the buildpacks, stacks, etc. are also part of the repository, that makes listing all of the top level directories for the `-l` flag as I suggested above, a little problematic. Should we also propose in the documentation for this, that language pack providers/vendors should publish the repository with a specific structure? For example, we could say that a certain directory name is reserved for these kinds of files - i.e. `builder`. The `func` CLI will ignore this directory when listing available languages for a repository, so vendors could use it to contain buildpacks and other related artifacts.

Reply:
Overall, I completely agree.  I do not, however, think the suggestions in any way conflict with the usefulness of this explicit "use this remote" flag, but rather build on it.  Perhaps the documentation will make that more clear, or prove me wrong when I try to express it therein.

Right now, it is a clear shortcoming that there is no feedback to the user which languages and templates are available within a repository (including the builtin), and thus we are dependent on good documentation.  This should be on our short list to improve.  Should this be part of this PR or a new one?

Additionally, creating a ligtweight repository management system (with requisite CLI tree where you can CRUD repo references), is indeed the general direction I was hoping we would go eventually!  That's why I went "local disk first".  What's the priority of this?  Should it even be part of this FT or a new one?

Reply:
For future reference,  a little thread housekeeping:

The relevant section from [the Feature Track Specification](https://hackmd.io/@j8e4sRzyQ2O1Gc5V3z7YBQ/BkIXfXC6_#Remote-Repositories-backlog)  to this PR states:

> Remote Repositories
> 
> In addition to built-in and on-disk repositories, func should also support referring to a template in a remote repository. When specifying a template in a remote repository, the location of the remote is a URL provided for the repository location, which is expected to be an available Git repository. For example, to use a TypeScript template provided by Microsoft in a GitHub repository named knative-fn a user will type:
> 
> `func create -l typescript -r https://github.com/microsoft/knative-fn `

This is the feature added by this PR.

The brainstorm above is mostly regarding the previous section in the FT which talks about local repository installation:

> Installing a Repository Locally
> 
> Installing a repository simply means the creation of a conformant file structure. To install a future Boson Project extended templates repository could be accomplished as follows:
> 
> `git clone https://github.com/boson-project/templates ~/.config/func/repositories/boson`
> 

This was completed earlier, and thus the majority of the discussion is about enhancement requests outside the scope of the original FT.    These enhancements are now being tracked in https://github.com/orgs/knative-sandbox/projects/4 and will appear in future PRs.